### PR TITLE
When running scan over a big repo, kics takes too much time to print the feedback to console

### LINKED ISSUE #1685: When running scan over a big repo, kics takes too much time to print the feedback to console, so user is not aware if it is running or not
### Expected Behavior
When initially we run kics, some information should be printed to the logs, so users are aware kics is doing something (no verbose mode) and it is not stuck, such as:
- Application Name
- Version
- Flags Provided
- Technologies Identified: Dockerfile, AWS, GCP...
- Scan Progress

For Example:
```
kics - Static Code Analysis of Infrastructure as Code.
@Checkmarx
Version: 1.1.0

Flags Provided:
-p | --path -> /path

Technologies Identified:
- Dockerfile

Total Files: 1000
Total IaC Identified Files: 1
Scanning 1 out of 1 files (100%)
Scan Success !
Results:
Files scanned: 1
Parsed files: 1
Queries loaded: 728
Queries failed to execute: 0
Healthcheck Instruction Missing, Severity: LOW, Results: 2
       Dockerfile:22
      Dockerfile:1
Missing USER Instruction, Severity: HIGH, Results: 2
      Dockerfile:1
      Dockerfile:22
Access Key Is Not Rotated Within 90 Days, Severity: MEDIUM, Results: 1
     Dockerfile:1
```

### Actual Behavior
Nothing is being printed before the scan is completed

### Steps to Reproduce the Problem

  1. git clone https://github.com/Checkmarx/kics
  2. Create a docker-compose.yml file:
```yaml
version: '3'

services:
  kics:
    platform: linux
    image: checkmarx/kics:latest
    container_name: kics
    hostname: kics
    volumes:
      - ./:/path/
    build:
      context: ./
      dockerfile: Dockerfile
    command: -p /path/
```
  4. Run docker-compose up
  5. Since kics Repo contains a lot of IAC files, kics only prints feedback to the console once the scan is finished, so user is not aware what is going on before and during the scan.

### Specifications

  - Version: latest
  - Platform: linux
  - Subsystem: docker

### PR DESCRIPTION
**Pull request by felipe-avelar**
Closes #1685 

**Proposed Changes**

- Added a progress bar showing how many queries already been executed
- Added a flag `--no-progress`, so user can silence progress bar

### CODE REVIEW

File: pkg/engine/inspector.go [hunk 1]
```diff
@@ -170,7 +178,18 @@ func (c *Inspector) Inspect(ctx context.Context, scanID string, files model.File
 	}
 
 	var vulnerabilities []model.Vulnerability
-	for _, query := range c.queries {
+	currentQuery := make(chan float64, 1)
+	var wg sync.WaitGroup
+	if !hideProgress {
+		wg.Add(1)
+		progressBar := consoleHelpers.NewProgressBar("Executed queries: ", 10, float64(len(c.queries)), currentQuery)
```
[the remark this answers is not in the document]
Comment (felipe-avelar) [reply pkg/engine/inspector.go:185]: I agree with you, gonna change to Executing instead Executed

**Review by joaoReigota1 — DISMISSED** [review]

**Review by joaoReigota1 — DISMISSED** [review]

**Review by rogeriopeixotocx — DISMISSED** [review]
LGTM

File: pkg/engine/inspector.go [hunk 2]
```diff
@@ -170,7 +178,18 @@ func (c *Inspector) Inspect(ctx context.Context, scanID string, files model.File
...
 
 	var vulnerabilities []model.Vulnerability
-	for _, query := range c.queries {
+	currentQuery := make(chan float64, 1)
+	var wg sync.WaitGroup
+	if !hideProgress {
+		wg.Add(1)
+		progressBar := consoleHelpers.NewProgressBar("Executing queries: ", 10, float64(len(c.queries)), currentQuery)
+		go progressBar.Start(&wg)
+	}
```
Comment (ruigomescx) [inline pkg/engine/inspector.go:187]: Can we encapsulate this into a method like startProgressBar()?

File: pkg/engine/inspector.go [hunk 3]
```diff
@@ -192,7 +211,11 @@ func (c *Inspector) Inspect(ctx context.Context, scanID string, files model.File
 
 		c.tracker.TrackQueryExecution()
 	}
-
+	if !hideProgress {
+		close(currentQuery)
+		wg.Wait()
+	}
+	fmt.Println("\r")
```
Comment (ruigomescx) [inline pkg/engine/inspector.go:218]: Can we apply the same principle as above?

**Review by rogeriopeixotocx — DISMISSED** [review]
LGTM

**Review by ruigomescx — APPROVED** [review]

**Pull request merged**